Sources/SCTE35/HelperConverter/BitConverter.swift

```diff
@@ -87,56 +166,168 @@ class BitConverter {
+
+        // Build ISAN string without check chars
+        var isanString = ""
+        for (index, int) in isanIntegers.enumerated() {
+            if index != 0 && index % 4 == 0 { isanString.append("-") }
+            let char = String(format: "%01X", int)
+            isanString.append(char)
+        }
+
+        // Add in check chars
+        if !containsVersion {
+            isanString.append("-\(firstCheckChar)")
+        } else {
+            let firstCheckDigitIndex = isanString.index(isanString.startIndex, offsetBy: 19)
+            isanString.insert(contentsOf: "-\(firstCheckChar)", at: firstCheckDigitIndex)
+            guard let theSecondCheckChar = secondCheckChar else { return nil }
+            isanString.append("-\(theSecondCheckChar)")
+        }
+
+        return isanString
+    }
+
+    static func bits(fromIsan isan: String) -> [Bit]? {
+        let containsVersion: Bool
+        if isan.count == 33 {
+            containsVersion = true
+        } else if isan.count == 21 {
+            containsVersion =  false
+        } else {
+            // character count does not meet requirements for ISAN string
+            return nil
+        }
+
+        // remove check digits from isan string
+        var isanHexDecimals = isan.replacingOccurrences(of: "-", with: "").dropLast() // dropLast removes the 2nd check digit
+        if containsVersion {
+            let firstCheckDigitIndex = isanHexDecimals.index(isanHexDecimals.startIndex, offsetBy: 16)
+            isanHexDecimals.remove(at: firstCheckDigitIndex)
+        }
+
+        var bits = [Bit]()
+        for hexVal in isanHexDecimals {
+            guard let alphaNumCharVal = ISO7064.alphanumericCharacterSet.firstIndex(of: hexVal) else { return nil }
+            bits.append(contentsOf: BitConverter.bits(from: alphaNumCharVal, bitArraySize: 4))
+        }
+
+        return bits
     }
 
     static func eidrString(fromBits bits: [Bit]) -> String? {
         guard bits.count == 96 else { return nil }
         let firstTwoBytes = Array(bits[0..<16])
-        let integerValue = BitConverter.integer(fromBits: firstTwoBytes)
-        let nibbles = stride(from: 16, to: bits.count, by: 4).map { (last) -> [Bit] in
-            return Array(bits[last..<last+4])
+        let subPrefix = BitConverter.integer(fromBits: firstTwoBytes)
+
+        var eidrIntegers = [Int]()
+        for index in stride(from: bits.startIndex.advanced(by: 16), to: bits.endIndex, by: 4) {
+            let nibble = Array(bits[index..<index+4])
+            let intValue = BitConverter.integer(fromBits: nibble)
+            eidrIntegers.append(intValue)
         }
 
-        var stringToReturn = "10.\(integerValue)/"
-        let indicesForDash = Set<Int>(arrayLiteral: 3, 7, 11, 15)
-        for (index, nibble) in nibbles.enumerated() {
+        guard let checkChar = ISO7064.calculateMod3637CheckCharacter(fromDecimals: eidrIntegers) else { return nil }
```

Comment:
EIDR check character is now calculated and included in the returned EIDR string.